docs/source/references/modules.rst

```diff
@@ -269,4 +262,60 @@ R2Score
 ~~~~~~~
 
 .. autoclass:: torchmetrics.R2Score
-    :noindex:
\ No newline at end of file
+    :noindex:
+
+*********
+Retrieval
+*********
+
+Input details
+~~~~~~~~~~~~~
+
+For the purposes of retrieval metrics, inputs (indexes, predictions and targets) must have the same size
+(``N`` stands for the batch size) and the following types:
+
+.. csv-table::
+    :header: "indexes shape", "indexes dtype", "preds shape", "preds dtype", "target shape", "target dtype"
+    :widths: 10, 10, 10, 10, 10, 10
+
+    "``long``", "(N,...)", "``float``", "(N,...)", "``long`` or ``bool``", "(N,...)"
+
+.. note::
+    All dimensions are "squeezed out" at the beginning, so
+    that, for example, a tensor of shape ``(N, M)`` is treated as ``(N * M, )``.
+
+In Information Retrieval you have a query that is compared with a variable number of documents. For each pair ``(Q_i, D_j)``,
```

Comment:
```suggestion
In information retrieval you have a query that is compared with a variable number of documents. For each pair ``(Q_i, D_j)``,
```

Reply:
I think that this can stay if we follow shortcut like "In Information Retrieval (IR) you have a query"